.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,57 @@
+
+#### API Specification
+You can find our API specification [here](https://documenter.getpostman.com/view/12949536/TVRedVwQ). It'll tell you about all the endpoints our API exposes and what each does. All the types in our specification, e.g. requests, responses, user profiles, merch orders, are defined in the `types` folder, and if you're adding a route or modifying an existing one, you'll first need to edit these types.
+
+#### API Implementation
+At a high level, our API is implemented in three layers: the endpoints layer, the services layer, and the data layer. The endpoints layer calls the services layer and the servies layer calls the data layer.
+
+##### Endpoints
+If you make a request to some endpoint, that request will be received in our endpoints layer, which handles anything that has to do with our routing logic, with requests and responses. This includes
++ executing middleware (e.g. verifying the authentication token that comes attached to most requests),
++ validating requests (ensuring that the request is formatted as required by our API specification, that no fields are missing, that numeric fields are numbers and not strings, etc.),
++ checking permissions,
++ calling the services layer with any relevant request data,
++ and sending back the response.
+
+This layer lives inside our `api` folder. In the unlikely event you'll need to write middleware, that'll go in the `middleware` folder. The `controllers` folder is where the bulk of our routing logic lives; within each controller, you'll see the API routes corresponding to that controller, e.g. `UserController` holds the routes under `/api/user`. Most of these routes require some sort of request body&mdash;the same request bodies you specified in the `types` folder&mdash;, and these request bodies need to be validated; to do so, create classes in the `validators` folder implementing the interfaces from the `types` folder. That leaves the actual endpoint code in the controllers, which should take any relevant data from the requests and call the services layer. Occasionally, requests have to be further validated in the controller, such as to ensure there're no duplicate or out-of-order elements in an array.
+
++ [Express](https://github.com/expressjs/express) is a web framework we use
++ [routing-controllers](https://github.com/typestack/routing-controllers) is a library we use on top of Express
++ [class-validator](https://github.com/typestack/class-validator) is a library we use for decorator-based validation
+
+##### Services
+The services layer is made up of all the actual "business logic" of the portal, the rules of the system (e.g. "a user can only attend an event once"). Each service is some grouping of functionality; for example, the `EventService` contains all code for creating events, reading events, etc. All our services can be found in the `services` folder.
```

Comment:
nit: 
> ... "business logic" of the portal, the rules of the system ..."

might be a fragment? I think replacing comma with "and" makes more sense